packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,600 @@
+PanelsWrapper.displayName = 'Panels.Wrapper';
+
+const PanelsWrapperContext = React.createContext<
+  | {
+      activeId?: PanelsWrapperProps['activeId'];
+      currentPanelId: CurrentPanelId;
+      triggers: React.RefObject<
+        Map<string, { triggerId: string; panelId: string }>
+      >;
+      changeActivePanel: (props: {
+        newActiveId: CurrentPanelId;
+        ignoreControlledMode?: boolean;
+      }) => Promise<void>;
+      goBack: PanelsInstance['goBack'];
+      panelElements: React.RefObject<Record<string, HTMLElement | null>>;
+      animations: Record<string, React.CSSProperties[]> | undefined;
+    }
+  | undefined
+>(undefined);
+
+// ----------------------------------------------------------------------------
+
+type PanelProps = {
+  id: string;
+};
+
+/**
+ * Takes an `id` and the panel content. When the wrapper's `activeId` is set to this panel id, this panel is shown.
+ *
+ * @example
+ * <Panels.Panel id={panelIdRoot} as={List}>
+ *   <Surface.Header as={Panels.Header}>Root</Surface.Header>
+ *   <ListItem>
+ *     <Panels.Trigger for={panelIdMoreInfo}>
+ *       <ListItem.Action>More details</ListItem.Action>
+ *     </Panels.Trigger>
+ *   </ListItem>
+ * </Panels.Panel>
+ */
+const Panel = React.forwardRef((props, forwardedRef) => {
+  const { id: idProp, children, className, style, ...rest } = props;
+
+  const { currentPanelId, triggers, panelElements, animations } =
+    React.useContext(PanelsWrapperContext) || {};
+
+  const fallbackId = React.useId();
+  const id = idProp || fallbackId;
+
+  const [element, setElement] = React.useState<HTMLElement | null>(null);
+  if (panelElements?.current != null) {
+    panelElements.current[id] = element;
+  }
+
+  const isAnimating = animations != null;
+  const isHidden =
+    animations == null
+      ? id !== currentPanelId?.id
+      : !Object.keys(animations).includes(id);
+
+  const isInert = isHidden || isAnimating;
+
+  const associatedTrigger = !!id ? triggers?.current?.get(id) : undefined;
+
+  return (
+    <PanelContext.Provider
+      value={{
+        id,
+        associatedTrigger,
+      }}
+    >
+      <Box
+        ref={useMergedRefs(setElement, forwardedRef)}
+        id={id}
+        aria-labelledby={`${id}-header`}
+        tabIndex={-1}
+        {...rest}
+        className={cx('iui-panel', className, {
+          'iui-panel-visible': id === currentPanelId?.id,
+        })}
+        style={{
+          display: isHidden ? 'none' : undefined,
+
+          // Add the last keyframe styles to the current panel to avoid flickering
+          // i.e. showing the current panel for a split second between when the animations ends and the panel is hidden
+          ...(id === currentPanelId?.id &&
+          animations != null &&
+          animations[id] != null
+            ? animations[id][animations[id].length - 1]
+            : {}),
+
+          ...style,
+        }}
+        {...{ inert: isInert ? '' : undefined }}
```

Comment:
We need to also load the polyfill for `inert` attribute.

Also, in the description you said "When not animating, all panels except the non-active panel are marked as inert", which I'm not sure is accurate? Could you try to rephrase how/when `inert` and `display` (or `hidden`) are applied so I understand it better?

Reply:
> When not animating: All `hidden` panels will be made `inert`

This is redundant. `inert` does nothing on `hidden` elements, it's only useful on visible elements.

> When animating: _All_ panels are made inert. This prevents clicks in the panels during the animation

I think we should keep the triggering panel non-inert. The main reason for this is so that the user's focus doesn't randomly get lost; it should stay on the button that they clicked on. The animation shouldn't take so long that the user has a chance to accidentally click on something else in the panel. After the animation is complete, the old panel can be `hidden`.